### PR TITLE
Upgrade Istio to 1.2.0

### PR DESCRIPTION
**Pull request by loganhz**
Upgrade Istio to 1.2.0

Related issue:

https://github.com/rancher/rancher/issues/20962

Need to push the images first

```
docker pull istio/kubectl:1.2.0
docker tag istio/kubectl:1.2.0 rancher/istio-kubectl:1.2.0
docker push rancher/istio-kubectl:1.2.0

docker pull istio/proxy_init:1.2.0
docker tag istio/proxy_init:1.2.0 rancher/istio-proxy_init:1.2.0
docker push rancher/istio-proxy_init:1.2.0

docker pull istio/node-agent-k8s:1.2.0
docker tag istio/node-agent-k8s:1.2.0 rancher/istio-node-agent-k8s:1.2.0
docker push rancher/istio-node-agent-k8s:1.2.0

docker pull istio/mixer:1.2.0
docker tag istio/mixer:1.2.0 rancher/istio-mixer:1.2.0
docker push rancher/istio-mixer:1.2.0

docker pull istio/citadel:1.2.0
docker tag istio/citadel:1.2.0 rancher/istio-citadel:1.2.0
docker push rancher/istio-citadel:1.2.0

docker pull istio/galley:1.2.0
docker tag istio/galley:1.2.0 rancher/istio-galley:1.2.0
docker push rancher/istio-galley:1.2.0

docker pull istio/proxyv2:1.2.0
docker tag istio/proxyv2:1.2.0 rancher/istio-proxyv2:1.2.0
docker push rancher/istio-proxyv2:1.2.0

docker pull istio/sidecar_injector:1.2.0
docker tag istio/sidecar_injector:1.2.0 rancher/istio-sidecar_injector:1.2.0
docker push rancher/istio-sidecar_injector:1.2.0

docker pull istio/pilot:1.2.0
docker tag istio/pilot:1.2.0 rancher/istio-pilot:1.2.0
docker push rancher/istio-pilot:1.2.0

docker pull grafana/grafana:6.1.6
docker tag grafana/grafana:6.1.6 rancher/grafana-grafana:6.1.6
docker push rancher/grafana-grafana:6.1.6 

docker pull kiali/kiali:v0.20
docker tag kiali/kiali:v0.20 rancher/kiali-kiali:v0.20
docker push rancher/kiali-kiali:v0.20

docker pull prom/prometheus:v2.8.0
docker tag prom/prometheus:v2.8.0 rancher/prom-prometheus:v2.8.0
docker push rancher/prom-prometheus:v2.8.0
```

### CODE REVIEW

File: charts/rancher-istio/0.0.1/Chart.yaml [hunk 1]
```diff
@@ -1,9 +1,10 @@
 apiVersion: v1
-name: istio
-version: 1.1.0
-appVersion: 1.1.0
+name: rancher-istio
+version: 0.0.1
+appVersion: 1.1.5
```
Comment (guangbochen) [inline charts/rancher-istio/0.0.1/Chart.yaml]: set appVersion to `1.2.0`

File: charts/rancher-istio/0.0.1/charts/certmanager/values.yaml [hunk 2]
```diff
@@ -4,12 +4,14 @@
 # gateway must be updated by adding 'secretVolumes'. After the gateway
 # restart, DestinationRules can be created using the ACME-signed certificates.
 enabled: false
+replicaCount: 1
 image:
   hub: quay.io
```
Comment (guangbochen) [inline charts/rancher-istio/0.0.1/charts/certmanager/values.yaml]: should remove `image.hub` as we are using `rancher/jetstack-cert-manager-controller` from dockerhub, 
also change the `image: {{ .Values.image.hub }}/{{ .Values.image.repository }}:{{ .Values.image.tag }}` to `image: {{ .Values.image.repository }}:{{ .Values.image.tag }}` in certmanager/templates/deployment.yaml  Line 41

**Review by guangbochen — CHANGES_REQUESTED** [review]

**Comment by loganhz** [comment]
@guangbochen pushed one change for your comments

**Comment by guangbochen** [comment]
LGTM.

**Comment by loganhz** [comment]
rebased

**Pull request merged**